src/medida/histogram.cc

```diff
@@ -146,18 +150,21 @@ void Histogram::Impl::Clear() {
 
 std::uint64_t Histogram::Impl::count() const {
   std::lock_guard<std::mutex> lock {mutex_};
-  return count_;
+  return sample_type_ == kCKMS ? sample_->MakeSnapshot().size() : count_;
```

Comment:
That makes a lot of sense. Initially, I thought that it might not hurt to report `min`, `max`, `mean` and `stddev` based on the previous time window, instead of all-time history, but I realized that it'd be really confusing to report some metrics based on all-time history and some metrics based on partial history.

I'll implement this by:

- Adding `max` to the `Snapshot` class since `CKMS` doesn't seem to be 100% accurate when it comes to the maximum value. Then I'll report the `max` value from the snapshot as `100%` in those 4 reporting methods.
- Removing the `min` `sum` `variance` that I added to the `Snapshot` class as it's no longer necessary.
- Similarly, I'll remove those histogram metric methods from `CKMS`.

This would simplify the implementation quite a bit. Also, I think this helps with migrating the change in Core. Some tests in Core seem to depend on the all-time `count` metric, and changing it to use the 30-second window would have probably broken those tests.

@MonsieurNicolas Just wanted to mention you here to keep you in the loop.

Reply:
Done. This simplified the change quite a bit :)

Reply:
For some reason, I completely forgot to update the reporting functions. I just did and pushed those changes here.